src/librustc_codegen_llvm/back/link.rs

```diff
@@ -212,12 +212,9 @@ fn link_binary_output(sess: &Session,
 }
 
 fn archive_search_paths(sess: &Session) -> Vec<PathBuf> {
-    let mut search = Vec::new();
-    sess.target_filesearch(PathKind::Native).for_each_lib_search_path(|path, _| {
-        search.push(path.to_path_buf());
-    });
-
-    search
+    sess.target_filesearch(PathKind::Native).search_paths()
+        .map(|sp| sp.dir.to_path_buf())
+        .collect()
```

Comment:
Literally the same thing, maybe the other method should be renamed to be just `search_path_dirs`.

Reply:
Sorry, I don't understand. Literally the same thing as what? What other method should be renamed?

Reply:
The one I commented one about `-> impl Iterator`.